### PR TITLE
Adds Beaker docker testing to .travis.yml

### PR DESCRIPTION
**Pull request by petems**
* Adds CentOS 6 Beaker docker file
* Adds travis matrix entry to run beaker-docker tests

### CODE REVIEW

**Comment by jhoblitt** [comment]
Is beaker officially compatible with ruby 2.2 now?

**Comment by petems** [comment]
I believe has for a while according to https://tickets.puppetlabs.com/browse/BKR-537

**Comment by petems** [comment]
It looks like the issue is with beaker-rspec and some versions rspec, this seems to be the workaround: https://github.com/puppetlabs/puppetlabs-firewall/pull/597/files

**Comment by petems** [comment]
To simplify things I switched it back to 2.1.0 anyways

**Comment by jhoblitt** [comment]
Awww, the ticket likely explains the failure I've seen.  This module isn't pinning the rspec version so it should, in theory, be able to work.

**Comment by jhoblitt** [comment]
Tangentially, https://github.com/jenkinsci/puppet-jenkins/pull/473 fixes the unit tests under puppet 4.2+.

**Comment by petems** [comment]
Cool, was gonna say, those tests take ages and 1.9.3 is EOL now :smile:

**Comment by jhoblitt** [comment]
I thought #473 has already been merged. so I just put it in.  It probably causes a merge conflict in the `.travis.yaml`.  My bad. ;(

1.9.3 is in the matrix as this is still the low bar for puppet 4.

**Comment by jhoblitt** [comment]
It looks like travis lost a worker mid-build.  I have restarted the job.

File: .travis.yml [hunk 1]
```diff
@@ -15,5 +15,15 @@ env:
   - PUPPET_VERSION="~> 4.1.0"
   - PUPPET_VERSION="~> 4.2.0"
   - PUPPET_VERSION="~> 4.3.0"
+matrix:
+  include:
+    - rvm: '2.1'
+      sudo: required
+      services: docker
+      env: PUPPET_VERSION="3.8.4" BEAKER_set="centos-6-docker"
+      script: bundle exec rspec spec/acceptance/class_spec.rb
```
Comment (jhoblitt) [inline .travis.yml]: Why not run all of (our vast collection...) of acceptance tests?

Comment (petems) [inline .travis.yml]: I was mainly worried about the time it would take, but I can change it to run everything if required? :smile:

Comment (petems) [inline .travis.yml]: Fixed :+1:

**Comment by petems** [comment]
Travis died mid build again :/

**Comment by jhoblitt** [comment]
I've kicked it again.  If this keeps up, we may need to stand up a jenkins instance ;)

**Comment by jhoblitt** [comment]
And I've kicked it again... what is going on?

**Comment by rtyler** [comment]
@jhoblitt I'm not against spinning this up in the [ci.jenkins-ci.org](https://ci.jenkins-ci.org) instance since it is fairly "official"

Should I file a ticket for that?

**Comment by jhoblitt** [comment]
@rtyler That certainly wouldn't hurt!  From looking at the other jobs, it appears that the GitHub Pull Request Builder plugin is already installed.

**Comment by petems** [comment]
@jhoblitt Does it make sense to remove the 1.9.3 tests if they're causing the issues? https://github.com/jenkinsci/puppet-jenkins/issues/509

**Comment by jhoblitt** [comment]
@petems I suspect we may have to but note that puppet 4.x is technically supported on 1.9.3 and we are not using the AIO packages with travis.

**Comment by petems** [comment]
@jhoblitt Can we move the 1.9.3 tests into allow_failure?

**Comment by rtyler** [comment]
@petems 1.9.3 is still the default Ruby packaged up for a lot of people, myself included, running not-the-latest-versions of PE.

The error with the build in Travis isn't a segfault but something stopped outputting anything which to me means that somebody went either into an infinite loop or prompted for something on `stdin`.

I do not wish to remove 1.9.3. testing plz

**Comment by petems** [comment]
Ok, will kill the issue then! :+1:

**Comment by jhoblitt** [comment]
It looks like the ruby 1.9.3 build times have pretty consistently been longer than ruby 2.1.0.  Here is a build from ~4 months ago. 

https://travis-ci.org/jenkinsci/puppet-jenkins/builds/86419131

Puppet 4.3.0 is just a tad slower than previous puppet versions, which is why we are all of a sudden hitting the travis 10min output limit.  However, at least once I have seen other ruby 1.9.3 combos fail as well.  We need to identify what test is taking slow long to execute.  As an interm, I think it would be reasonable to exclude the ruby 1.9.3/puppet 4.3 combo from the matrix.

Since I have convinced myself that the failure is not related to anything in this PR (it is also happening in #508), I'm going to go ahead and merge this.

@petems - I really appreciate the time you have put into this PR.  This module is in desperate need of acceptance tests that cover the jenkins<->puppet interface that rspec-puppet can't address.

**Pull request merged**